geomstats/geometry/lie_group.py

```diff
@@ -17,10 +29,74 @@ class MatrixLieGroup(Manifold, abc.ABC):
     """Class for matrix Lie groups."""
 
     def __init__(self, representation_dim, lie_algebra=None, **kwargs):
-        super().__init__(shape=(representation_dim, representation_dim), **kwargs)
         self.lie_algebra = lie_algebra
+        super().__init__(shape=(representation_dim, representation_dim), **kwargs)
         self.representation_dim = representation_dim
 
+    def _equip_lie_algebra_for_invariant_metric(self):
+        """Equip Lie algebra with appropriate metric.
+
+        Three cases for metric on Lie algebra:
+        * no `metric_mat_at_identity` or diagonal with ones: MatricesMetric
+        * diagonal `metric_mat_at_identity`: MatricesDiagMetric
+        * else: generic metric_matric on an Euclidean space
+        """
+        lie_algebra = self.lie_algebra
+        if self.metric.metric_mat_at_identity is None:
+            lie_algebra.equip_with_metric(MatricesMetric)
+            return self
+
+        if Matrices.is_diagonal(self.metric.metric_mat_at_identity):
+            metric_coeffs_vec = gs.diagonal(self.metric.metric_mat_at_identity)
+            if gs.amax(gs.abs(metric_coeffs_vec - 1.0)) < gs.atol:
+                lie_algebra.equip_with_metric(MatricesMetric)
+                return self
+
+            lie_algebra.equip_with_metric(
+                MatricesDiagMetric,
+                metric_coeffs=gs.abs(
+                    lie_algebra.matrix_representation(metric_coeffs_vec)
+                ),
+            )
+            return self
+
+        basis_repr_diffeo = BasisRepresentationDiffeo(lie_algebra)
+        lie_algebra.equip_with_metric(
+            PullbackDiffeoMetric,
+            diffeo=basis_repr_diffeo,
+            image_space=Euclidean(lie_algebra.dim, equip=False).equip_with_metric(
+                FlatRiemannianMetric, metric_matrix=self.metric.metric_mat_at_identity
+            ),
+        )
+        return self
+
+    def _equip_lie_algebra_for_biinvariant_metric(self):
```

Comment:
Some Lie groups do not admit any bi-invariant metric. Thus, I'm not sure what this function is really doing?

Only compact group, abelian groups, and their products, admit bi-invariant metrics. For example, SE(n) never admits any bi-invariant metric.